main.cpp

```diff
@@ -269,19 +379,34 @@ struct NoiseGen
     double sahControlLFORate( double sahControl, int routingSAH = 2  );
 };
 
+NoiseGen::NoiseGen()
+{
+    noiseColourPink = false;
+    noiseGain = 0.0;
+    sahRate = 8.0;
+    sahControl = 0.0; 
+    routingSAH = 0;
+    std::cout << "Instance of 'NoiseGen' has been constructed" << std::endl;
+}
+
 struct Synth
 {
-    Oscillator oscillator; 
-    LFO lfo;
-    Envelope envelope; 
-    MixerAndMidi MixerAndMidi;
-    NoiseGen noiseGen;
+    Synth();
 
     double bassTone();
     double snareDrum();
     int sendMidiNote();
 };
 
+Synth::Synth()
+{
+    Oscillator oscillator; 
+    LFO lfo;
+    Envelope myEnvelope(50,200,0.5,1000); 
+    MixerAndMidi MixerAndMidi;
+    NoiseGen noiseGen;
+}
```

Comment:
Now your synth has no member variables.
- [ ] revert

add space after `,` to make the code more readable.
- [ ] revise